webapp/src/components/assignee_modal/assignee_form.jsx

```diff
@@ -0,0 +1,156 @@
+import React, {useState, useEffect, useCallback} from 'react';
+import PropTypes from 'prop-types';
+
+import AutocompleteSelector from '../user_selector/autocomplete_selector';
+import Button from '../../widget/buttons/button';
+import IconButton from '../../widget/iconButton/iconButton';
+
+import CompassIcon from '../icons/compassIcons';
+
+const AssigneeForm = (
+    {
+        visible,
+        close,
+        autocompleteUsers,
+        theme,
+        getAssignee,
+        removeAssignee,
+        removeEditingTodo,
+        changeAssignee,
+        editingTodo,
+    },
+) => {
+    const [assignee, setAssignee] = useState();
+
+    useEffect(() => {
+        function handleKeypress(e) {
+            if (e.key === 'Escape') {
+                close();
+            }
+        }
+
+        document.addEventListener('keyup', handleKeypress);
+
+        return () => {
+            document.removeEventListener('keyup', handleKeypress);
+        };
+    }, [visible]);
```

Comment:
Would be interesting to put this in its own hook for convenience

```ts
const useEscapeKey = (close: () => void) => {
    useEffect(() => {
        function handleKeypress(e) {
            if (e.key === 'Escape') {
                close();
            }
        }

        document.addEventListener('keyup', handleKeypress);

        return () => {
            document.removeEventListener('keyup', handleKeypress);
        };
    }, [close]);
};
```

then in the component

```ts
    useEscapeKey(close);
```